src/supervisor/patch/0001-Prevent-supervisorctl-start-from-hanging-if-system-t.patch

```diff
@@ -0,0 +1,28 @@
+From 9ce7c3273d78dc412e3a1e3c95307a63846bad39 Mon Sep 17 00:00:00 2001
+From: Joe LeVeque <jolevequ@microsoft.com>
+Date: Tue, 16 Jan 2018 20:42:13 +0000
+Subject: [PATCH] Prevent 'supervisorctl start' from hanging if system time
+ rolls backward
+
+---
+ supervisor/process.py | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/supervisor/process.py b/supervisor/process.py
+index f9ddcd9..d292421 100644
+--- a/supervisor/process.py
++++ b/supervisor/process.py
+@@ -588,6 +588,10 @@ class Subprocess:
+ 
+         logger = self.config.options.logger
+ 
++        # If system clock has moved backward, reset self.laststart to current system time
++        if now < self.laststart:
++            self.laststart = now;
```

Comment:
I am ok with this pariticular fix. however, I think the time travel can cause other problems in the code.

i checked the supervisord code, it looks like there are other places that save the current time to a value. like self.delay, it could also have the same problem in the back-off process.

if you are in the backoff state, and the system clock moved backward, then you need wait for a long time to catch up and the process won't restart for a long time.

another example is last_dispatch.

this does not sound like a trivial problem. I also see there are some unittest added. I think we need to fix all the problems I see above and add unit test code and add unit test code.